src/averageAges.js

```diff
@@ -20,6 +20,14 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const male = (person) => (person.sex === 'm');
+  const didInCentury
+    = (person) => (male(person) && (Math.ceil(person.died / 100) === century));
+  const result
+    = people.filter(century ? didInCentury : male);
+
+  return result.reduce((prev, year) =>
+    prev + (year.died - year.born), 0) / result.length;
```

Comment:
but better to save the result of reduce to a variable and use it here 

```suggestion
  return result.reduce((prev, year) => (
      prev + (year.died - year.born)
    ), 0) / result.length;
```